### PR TITLE
Numerically stabilize autocorrelation()

### PR DESCRIPTION
**Pull request by fritzo**
This sets the autocorrelation of constant series to be 1.0 rather than NAN.

I believe this is the proper behavior for MCMC testing, where series with no variation should be seen as having effective sample size of 1.

My use case is in computing autocorrelation of discrete time series, where I'm using autocorrelation as a diagnostic to detect lack of mixing.  In that case I believe this PR is correct.

## Tested
- [x] added two unit tests
- [x] avoided NANs locally in [this notebook](https://github.com/fritzo/notebooks/blob/master/potts-locally-informed-mh.ipynb)

### CODE REVIEW

**Review by martinjankowiak — COMMENTED** [review]
is this the right fix? under what circumstances does `autocorr[..., :1]` get small?

**Comment by fritzo** [comment]
Yeah I don't know if this is the right fix.  The issue is when one value doesn't vary at all, so in that case I think the autocorrelation should be 1.

**Comment by martinjankowiak** [comment]
would it make sense to e.g. add an explicit test that checks that the resulting effective sample size is small for a variable with zero variance?

**Comment by fritzo** [comment]
OK I've added some tests.

> add an explicit test that checks that the resulting effective sample size is small for a variable with zero variance

hmm, it looks like our `effective_sample_size()` function suffers from the same problem because it [divides by the variance](https://github.com/pyro-ppl/pyro/blob/dc1351a8d44581aca8e32a29939c81777e981c2e/pyro/ops/stats.py#L188) which may be zero.

File: tests/ops/test_stats.py [hunk 1]
```diff
@@ -114,6 +114,27 @@ def test_autocorrelation():
     )
 
 
+def test_autocorrelation_trivial():
+    x = torch.zeros(10)
+    with xfail_if_not_implemented():
```
Comment (martinjankowiak) [inline tests/ops/test_stats.py:119]: i guess this xfail is no longer required?

**Review by martinjankowiak — DISMISSED** [review]

File: tests/ops/test_stats.py (hunk 1 above)
Comment (fritzo) [reply tests/ops/test_stats.py:119]: sure, let me try removing it :+1:

**Review by martinjankowiak — APPROVED** [review]

**Comment by fritzo** [comment]
Thanks for reviewing!
